fabric/v4/api_routing_protocols.go

```diff
@@ -29,37 +30,34 @@ var (
 
 type RoutingProtocolsApiService service
 
+func (v *RoutingProtocolData) UnmarshalJSON(data []byte) error {
+	strData := string(data)
+	if strData == "null" || strData == `""` {
+		return nil
+	}
 
-func (a *RoutingProtocolsApiService) decodeRP(v *RoutingProtocolData, b []byte, contentType string) (err error) {
-	//get type of response body
 	var (
-		localVarReturnBGP    RoutingProtocolBgpData
-		localVarReturnDirect RoutingProtocolDirectData
+		d RoutingProtocolDirectData
+		b RoutingProtocolBgpData
 	)
 
-	err = a.client.decode(&v, b, contentType)
-	if err != nil {
-		return err
-	}
-	// decode response based on type, overwrite return value's oneof field
-	if v.Type_ == "BGP" {
```

Comment:
What is the issue being addressed here?  Is there something we could change in the spec so that this gets generated correctly, or does it have to be done by modifying the generated code?

Reply:
Very difficult to change in the spec. Fabric's spec is using a OneOf keyword to indicate that the returned data could be one of two different models. Because of that the unmarshaller has difficulty populated the embedded struct fields for go. This is a custom implementation of the Unmarshaller for that model contain the OneOf embedded struct chain to allow bulk response and single responses to actually deserialize the data into the correct structs and let the models be usable.